src/modules/previewpane/powerpreview/dllmain.cpp

```diff
@@ -0,0 +1,233 @@
+            L"Show Markdown", 
+            PowerPreviewSettings.mdPreview_in_prevpane_IsEnabled);
+
+        return settings.serialize_to_buffer(buffer, buffer_size);
+    }
+
+    // Called by the runner to pass the updated settings values as a serialized JSON.
+    virtual void set_config(const wchar_t* config) override
+    {
+        try
+        {
+            // Parse the input JSON string.
+            PowerToysSettings::PowerToyValues values =
+                PowerToysSettings::PowerToyValues::from_json_string(config);
+
+            // Update settings for SVG icon render in Windows File Explorer. 
+            auto explr_rendrsvg_bool_toggle = values.get_bool_value(PowerPreviewSettings.EXPLR_RENDRSVG_BOOL_TOGGLE);
+            PowerPreviewSettings.svgPreview_in_explr_IsEnabled = explr_rendrsvg_bool_toggle.value();
+            if (PowerPreviewSettings.svgPreview_in_explr_IsEnabled)
+            {
+                Trace::ExplorerSVGRenderEnabled();
+            }
+            else
+            {
+                Trace::ExplorerSVGRenderDisabled();
+            }
+
+            // Update settings for SVG file render in Preview Pane.
+            auto prevpane_rendrsvg_bool_toggle = values.get_bool_value(PowerPreviewSettings.PREVPANE_RENDRSVG_BOOL_TOGGLE);
+            PowerPreviewSettings.svgPreview_in_prevpane_IsEnabled = prevpane_rendrsvg_bool_toggle.value();
+            if (PowerPreviewSettings.svgPreview_in_prevpane_IsEnabled)
+            {
+                Trace::PreviewPaneMarkDownRenderEnabled();
+            }
+            else
+            {
+                Trace::PreviewPaneMarkDownRenderDisabled();
+            }
+
+            // Update settings for Mark Down render in preview pane.
+            auto prevpane_rendrmd_bool_toggle = values.get_bool_value(PowerPreviewSettings.PREVPANE_RENDMD_BOOL_TOGGLE);
+            PowerPreviewSettings.mdPreview_in_prevpane_IsEnabled = prevpane_rendrmd_bool_toggle.value();
+            if (PowerPreviewSettings.mdPreview_in_prevpane_IsEnabled)
+            {
+                Trace::PreviewPaneMarkDownRenderEnabled();
+            }
+            else
+            {
+                Trace::PreviewPaneMarkDownRenderDisabled();
+            }
+
+            values.save_to_settings_file();
+        }
+        catch (std::exception const& e)
+        {
+            Trace::SetConfigInvalidJSON(e.what());
+        }
+    }
+
+    // Enable the powertoy
+    virtual void enable()
+    {
+        m_enabled = true;
+        Trace::FilePreviewerIsEnabled();
+    }
+
+    // Disable the powertoy
+    virtual void disable()
+    {
+        m_enabled = false;
+        Trace::FilePreviewerIsDisabled();
+    }
+
+    // Returns if the powertoys is enabled
+    virtual bool is_enabled() override
+    {
+        return m_enabled;
+    }
+
+    // Handle incoming event, data is event-specific
+    virtual intptr_t signal_event(const wchar_t* name, intptr_t data) override
+    {
+        return 1;
```

Comment:
This is `return 0`.  Return `1` is to suppress the event. Even though we are subscribing for any hook but default should be 0.